nixos/modules/services/torrent/qbittorrent-nox.nix

```
@@ -0,0 +1,141 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.qbittorrent-nox;
+in {
+  options = {
+    services = {
+      qbittorrent-nox = {
+        enable = mkEnableOption (lib.mdDoc "qbittorrent-nox daemon");
+
+        web = {
+          port = mkOption {
+            type = types.port;
+            default = 8080;
+            description = lib.mdDoc ''
+              qbittorrent-nox web UI port.
+            '';
+          };
+          openFirewall = mkOption {
+            type = types.bool;
+            default = false;
+            description = lib.mdDoc ''
+              Whether to open the firewall for the ports in
+              {option}`services.qbittorrent-nox.web.port`.
+            '';
+          };
+        };
+
+        torrenting = {
+          port = mkOption {
+            type = types.port;
+            default = 48197;
+            description = lib.mdDoc ''
+              qbittorrent-nox web UI port.
+            '';
+          };
+
+          openFirewall = mkOption {
+            default = false;
+            type = types.bool;
+            description = lib.mdDoc ''
+              Whether to open the firewall for the ports in
+              {option}`services.qbittorrent-nox.torrenting.port`.
+            '';
+          };
+        };
+
+        dataDir = mkOption {
+          type = types.path;
+          default = "/var/lib/qbittorrent-nox";
+          description = lib.mdDoc ''
+            The directory where qbittorrent-nox will create files.
+          '';
+        };
+
+        user = mkOption {
+          type = types.str;
+          default = "qbittorrent";
+          description = lib.mdDoc ''
+            User account under which qbittorrent-nox runs.
+          '';
+        };
+
+        group = mkOption {
+          type = types.str;
+          default = "qbittorrent";
+          description = lib.mdDoc ''
+            Group under which qbittorrent-nox runs.
+          '';
+        };
+
+        package = mkPackageOption pkgs "qbittorrent-nox" { };
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+
+    services.qbittorrent-nox.package = mkDefault (pkgs.qbittorrent-nox);
+
+    systemd.tmpfiles.rules = [
+      "d '${cfg.dataDir}' 0770 ${cfg.user} ${cfg.group}"
+      "d '${cfg.dataDir}/.config' 0770 ${cfg.user} ${cfg.group}"
+      "d '${cfg.dataDir}/.config/qBittorrent' 0770 ${cfg.user} ${cfg.group}"
+    ];
+
+    systemd.services.qbittorrent-nox = {
```

Same note about renaming the service

```suggestion
    systemd.services.qbittorrent = {
```